tests/test_patterns.py

```diff
@@ -146,8 +149,25 @@ def test_file_pattern_concat_merge(runtime_secrets, pickle, concat_merge_pattern
 
 
 @pytest.mark.parametrize("nkeep", [1, 2])
-def test_prune(nkeep, concat_merge_pattern):
-    fp = concat_merge_pattern[0]
+def test_prune(nkeep, concat_merge_pattern_with_kwargs, runtime_secrets):
+    if not concat_merge_pattern_with_kwargs:
+        # if `fsspec_open_kwargs` are passed with `is_opendap`, `FilePattern.__init__` raises
+        # ValueError and `concat_merge_pattern_with_kwargs` returns None, so nothing to test
+        return
```

Comment:
I'm not a fan of this behavior (having a fixture sometimes return `None`). It feels hacky and fragile. 

Best solution: refactor the fixtures so that this situation (`fsspec_open_kwargs` are passed with `is_opendap`) never happens.

Acceptable solution: make `concat_merge_pattern_with_kwargs` call `pytest.skip` when this condition is met.